sdk/src/beta9/runner/function.py

```diff
@@ -171,8 +172,9 @@ def main(channel: Channel):
         # Invoke the function and handle its result
         result = invoke_function(function_stub, context, task_id)
         if result.exception:
-            handle_task_failure(result, gateway_stub, task_id, container_id, container_hostname)
             thread_pool.shutdown(wait=False)
+            handle_task_failure(gateway_stub, result, task_id, container_id, container_hostname)
+            print(result.exception)
```

Comment:
cleanup

Reply:
We want this. This needs to be picked up by the log interceptor so the log gets a task id.